### PR TITLE
Added bb calculation function for layouts

### PR DESCRIPTION
**Pull request by mj3cheun**

### CODE REVIEW

File: documentation/md/collection/layoutPositions.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ var options = {
   eles: someCollection, // collection of elements involved in the layout; set by cy.layout() or eles.layout()
   fit: true, // whether to fit the viewport to the graph
   padding: 30, // padding to leave between graph and viewport
+  nodeDimensionsIncludeLabels: false, // Boolean which changes whether label dimensions are included when calculating node dimensions
```
Comment (maxkfranz) [inline documentation/md/collection/layoutPositions.md]: We need to document the `node.layoutDimensions()` function.  While technically the option could also be read in `layoutPositions()`, the `layoutPositions()` function doesn't use it.

File: documentation/docmaker.json [hunk 2]
```diff
@@ -1639,6 +1639,23 @@
                 }
               ],
               "md": "collection/layoutPositions"
+            },
+
+            {
+              "name": "nodes.layoutDimensions",
+              "descr": "Returns the node width and height",
```
Comment (maxkfranz) [inline documentation/docmaker.json]: Though it's important to be concise in descriptions, it's also important to be clear enough to give users a sense of how one function differs from another.  I think users might confuse this function with ones like `width()`, `height()`, and `boundingBox()`.  Maybe it's worth mentioning that it's for layout positioning (e.g. overlap detection) in the top description?

By the way, you can run `gulp docs` to see your spec changes reflected in the docs page.

**Comment by maxkfranz** [comment]
Great!  Thanks

**Comment by mj3cheun** [comment]
Note: Ensure external layouts are also updated to use the new function

**Comment by maxkfranz** [comment]
Updated w/ checklist : https://github.com/cytoscape/cytoscape.js/issues/1626

**Pull request merged**